### DISCUSSION TITLE
My config file does not seem to be read properly

### DISCUSSION THREAD

**Discussion by TechPro424** [falkreon/Jankson#74] [Q&A] [answered]
https://gist.github.com/TechPro424/8f26b39572f7274bac9a5e9b16c07d5e
I have a `send()` method in the `SaleItem` class that executes only if the config value `analyticsEnabled` is true.

I call the `JsonOperations.loadConfigFromFile()` method once in the mod's main entrypoint to ensure that the config is generated properly, which it is.

But when I change the `analyticsEnabled` value to false in the json5 file and start the mod, it still executes the `send()` function

Edit: I used my mod's `LOGGER` to check the value of the config field before the function executes, and it returns `true` even when the value of the field in the json5 file is `false`

**Comment by falkreon** [comment]
Couple notes here- for clarity, in `JsonOperations::loadConfigFromFile`, I'd make `config` a local variable, so you don't leak state into the method accidentally, and return from within the try..catch block so you explicitly control what is returned when the config load fails.

Any chance I can look at the json file with the analyticsEnabled field set to `false`? That's the last piece of the puzzle.

Either way, the problem is non-obvious. One possibility is that you're getting a thread-local version of Config::instance - there may be one on the client and one on the server, depending on when you classload the object. Looks like everything's happening on-server, which is the safest, but it's something worth looking into.

**Reply by TechPro424** [reply]
Implemented your suggestions, but the issue still persists
Here's the json file
```json
{ 
	"bidAmountIncrease": 50,
	"accountId": { 
		"namespace": "auction-house",
		"path": "auction-house-account"
	},
	"aliasesEnabled": true,
	"analyticsEnabled": false
}
```

**Reply by TechPro424** [reply]
I tried running the server and connecting to it with my main client, that way there's only one instance of the config, but the issue persists
So I don't think that's the issue

**Reply by TechPro424** [reply]
@falkreon bump

**Comment by TechPro424** [comment, marked as answer]
Fixed by updating jankson in build.gradle to v1.2.2
I was on v1.1.0 (that's what the GH wiki uses, it might need an update)
